contracts/finance/andromeda-weighted-distribution-splitter/src/contract.rs

```diff
@@ -0,0 +1,938 @@
+    // Amount of coins sent should be at least 1
+    require(
+        !sent_funds.is_empty(),
+        ContractError::InvalidFunds {
+            msg: "Require at least one coin to be sent".to_string(),
+        },
+    )?;
+    // Can't send more than 5 types of coins
+    require(
+        info.funds.len() < 5,
+        ContractError::ExceedsMaxAllowedCoins {},
+    )?;
+
+    let splitter = SPLITTER.load(deps.storage)?;
+    let mut msgs: Vec<SubMsg> = Vec::new();
+
+    let mut remainder_funds = info.funds.clone();
+
+    let mut total_weight = Uint128::zero();
+
+    // Calculate the total weight of all recipients
+    for recipient_addr in &splitter.recipients {
+        let recipient_weight = recipient_addr.weight;
+        total_weight += recipient_weight;
+    }
+
+    // Each recipient recieves the funds * (the recipient's weight / total weight of all recipients)
+    // The remaining funds go to the sender of the function
+    for recipient_addr in &splitter.recipients {
+        let recipient_weight = recipient_addr.weight;
+        let mut vec_coin: Vec<Coin> = Vec::new();
+        for (i, coin) in sent_funds.iter().enumerate() {
+            let mut recip_coin: Coin = coin.clone();
+            recip_coin.amount = coin.amount.multiply_ratio(recipient_weight, total_weight);
+            remainder_funds[i].amount -= recip_coin.amount;
+            vec_coin.push(recip_coin);
+        }
+        // ADO receivers must use AndromedaMsg::Receive to execute their functionality
+        // Others may just receive the funds
+        let msg = recipient_addr.recipient.generate_msg_native(
+            deps.api,
+            &deps.querier,
+            ADOContract::default().get_app_contract(deps.storage)?,
+            vec_coin,
+        )?;
+        msgs.push(msg);
+    }
+    remainder_funds = remainder_funds
+        .into_iter()
+        .filter(|x| x.amount > Uint128::zero())
+        .collect();
+
+    if !remainder_funds.is_empty() {
+        msgs.push(SubMsg::new(CosmosMsg::Bank(BankMsg::Send {
+            to_address: info.sender.to_string(),
+            amount: remainder_funds,
+        })));
+    }
+
+    Ok(Response::new()
+        .add_submessages(msgs)
+        .add_attributes(vec![attr("action", "send"), attr("sender", info.sender)]))
+}
+
+fn execute_update_recipients(
+    deps: DepsMut,
+    info: MessageInfo,
+    recipients: Vec<AddressWeight>,
+) -> Result<Response, ContractError> {
+    require(
+        ADOContract::default().is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+    // No need to send funds
+    require(
+        info.funds.is_empty(),
+        ContractError::FunctionDeclinesFunds {},
+    )?;
+
+    validate_recipient_list(recipients.clone())?;
+
+    let mut splitter = SPLITTER.load(deps.storage)?;
+    if splitter.locked {
+        StdError::generic_err("The splitter is currently locked");
+    }
```

Comment:
Should use `ContractError` instead. This check isn't actually doing anything since the error isn't being returned (note the `;` at the end).